docs/concepts/projects.md

```diff
@@ -0,0 +1,186 @@
+- [`prefect.yaml`](#the-prefect-yaml-file): a YAML file describing procedural steps for preparing a deployment from this project, as well as instructions for preparing the execution environment for a deployment run
+- [`./prefect`](#the-prefect-directory): a hidden directory where Prefect will store workflow metadata
+
+Projects can be initialized via the CLI command `prefect project init` run anywhere you consider to be the root of a project.  
+
+!!! tip "Project recipes"
+    Prefect ships with many off-the-shelf "recipes" that allow you to get started with more structure within your `deployment.yaml` and `prefect.yaml` files; run `prefect project recipe ls` to see what recipes are available in your installation. You can provide a recipe name in your initialization command with the `--recipe` flag, otherwise Prefect will attempt to guess an appropriate recipe based on the structure of your project directory (for example if you initialize within a `git` repository, Prefect will use the `git` recipe).
+
+## The Deployment YAML file
+
+The `deployment.yaml` file contains default configuration for all deployments created from within this project; all settings within this file can be overridden via the `prefect deploy` CLI command when creating a deployment.
+
+The base structure for `deployment.yaml` is as follows:
+
+```yaml
+# base metadata
+name: null
+version: null
+tags: []
+description: null
+schedule: null
+
+# flow-specific fields
+flow_name: null
+entrypoint: null
+path: null
+parameters: {}
+parameter_openapi_schema: null
+
+# infra-specific fields
+work_pool:
+  name: null
+  work_queue_name: null
+  job_variables: {}
+```
+
+You can create deployments via the CLI command `prefect deploy` without ever needing to alter this file in any way - its sole purpose is for version control and providing base settings in the situation where you are creating many deployments from your project.  [As described below](#deployment-mechanics), when creating a deployment these settings are first loaded from this base file, and then any additional flags provided via `prefect deploy` are layered on top before registering the deployment with the Prefect API.
+
+### Templating Options
+
+Values that you place within your `deployment.yaml` file can reference dynamic values in two different ways:
+
+- **step outputs**: every step of both `build` and `push` produce named fields such as `image_name`; you can reference these fields within `deployment.yaml` and `prefect deploy` will populate them with each call.  References must be enclosed in double brackets and be of the form `"{{ field_name }}"`
+- **blocks**: [Prefect blocks](/concepts/blocks) can also be referenced with the special syntax `{{ prefect.blocks.block_type.block_slug }}`
+
+As an example, consider the following `deployment.yaml` file:
+
+```yaml
+# base metadata
+name: null
+version: "{{ image_tag }}"
+tags:
+    - "{{ image_tag }}"
```

Comment:
I notice that we're encouraging use of templating here, but the code look like we only apply values to the `work_pool.job_variables` section. Is the plan to extend templating more broadly before release?

Reply:
Oh wow - great observation! Yea I will expand the templating logic to cover all of `deployment.yaml`